src/pytest_molecule/__init__.py

```diff
@@ -1,257 +1,287 @@
+            package
+        ).version
+
+    if "Tools" not in config._metadata:
+        config._metadata["Tools"] = {}
+    config._metadata["Tools"]["ansible"] = str(ansible_version())
+
+    # Adds interesting env vars
+    env = ""
+    for key, value in sorted(os.environ.items()):
+        for var_name in interesting_env_vars:
+            if key.startswith(var_name):
+                env += f"{key}={value} "
+    config._metadata["env"] = env
+
+    # We hide DeprecationWarnings thrown by driver loading because these are
+    # outside our control and worse: they are displayed even on projects that
+    # have no molecule tests at all as pytest_configure() is called during
+    # collection, causing spam.
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=DeprecationWarning)
+
+        config.option.molecule = {}
+        for driver in map(str, drivers()):
+            config.addinivalue_line(
+                "markers",
+                "{0}: mark test to run only when {0} is available".format(driver),
+            )
+            config.option.molecule[driver] = {"available": True}
+
+        config.addinivalue_line(
+            "markers",
+            "no_driver: mark used for scenarios that do not contain driver info",
+        )
+
+        config.addinivalue_line(
+            "markers", "molecule: mark used by all molecule scenarios"
+        )
+
+        # validate selinux availability
+        if sys.platform == "linux" and os.path.isfile("/etc/selinux/config"):
+            try:
+                import selinux  # noqa pylint: disable=unused-import,import-error,import-outside-toplevel
+            except ImportError:
+                logging.error(
+                    "It appears that you are trying to use "
+                    "molecule with a Python interpreter that does not have the "
+                    "libselinux python bindings installed. These can only be "
+                    "installed using your distro package manager and are specific "
+                    "to each python version. Common package names: "
+                    "libselinux-python python2-libselinux python3-libselinux"
+                )
+                # we do not re-raise this exception because missing or broken
+                # selinux bindings are not guaranteed to fail molecule execution.
+
+
+if int(pytest.__version__[0]) >= 7:
+
+    def pytest_collect_file(
+        parent: pytest.Collector, file_path: Optional["Path"]
+    ) -> Optional["Node"]:
+        """Transform each found molecule.yml into a pytest test."""
+        if file_path and file_path.is_symlink():
+            return None
+        if file_path and file_path.name == "molecule.yml":
+            return MoleculeFile.from_parent(path=file_path, parent=parent)
+        return None
+
+else:
+
+    def pytest_collect_file(parent: pytest.Collector, path) -> Optional["Node"]:

```

Comment:
mypy complains here with:

```
src/pytest_molecule/__init__.py:142: error: All conditional function variants must have identical signatures
```

@ssbarnea this is intentional since they actually are different in the pytest versions. Ideas? I tend to ignore the error but maybe you have a better idea for the implementation?

Thanks in advance!